trunk/SUAVE/Components/Energy/Converters/Combustor.py

```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
         self.outputs.fuel_to_air_ratio       = f 
     
+    def compute_rayleigh(self,conditions):
+        """ This computes the output values from the input values according to
+        equations from the source.
+
+        Assumptions:
+        Constant efficiency and pressure ratio
+
+        Source:
+        https://web.stanford.edu/~cantwell/AA283_Course_Material/AA283_Course_Notes/
+
+        Inputs:
+        conditions.freestream.
+          isentropic_expansion_factor         [-]
+          specific_heat_at_constant_pressure  [J/(kg K)]
+          temperature                         [K]
+          stagnation_temperature              [K]
+        self.inputs.
+          stagnation_temperature              [K]
+          stagnation_pressure                 [Pa]
+
+        Outputs:
+        self.outputs.
+          stagnation_temperature              [K]  
+          stagnation_pressure                 [Pa]
+          stagnation_enthalpy                 [J/kg]
+          fuel_to_air_ratio                   [-]
+
+        Properties Used:
+        self.
+          turbine_inlet_temperature           [K]
+          pressure_ratio                      [-]
+          efficiency                          [-]
+          area_ratio                          [-]
+          fuel_data.specific_energy           [J/kg]
+        """         
+        # unpack the values
+
+        # unpacking the values from conditions
+        gamma  = conditions.freestream.isentropic_expansion_factor 
+        Cp     = conditions.freestream.specific_heat_at_constant_pressure
+        To     = conditions.freestream.temperature
+        Tto    = conditions.freestream.stagnation_temperature
+        
+        # unpacking the values form inputs
+        Tt_in  = self.inputs.stagnation_temperature
+        Pt_in  = self.inputs.stagnation_pressure
+        Mach   = self.inputs.mach_number
+        Tt4    = self.turbine_inlet_temperature
+        pib    = self.pressure_ratio
+        eta_b  = self.efficiency
+        
+        # unpacking values from self
+        htf             = self.fuel_data.specific_energy
+        ar              = self.area_ratio
+        
+        # Rayleigh flow analysis, constant pressure burner
+            
+        # Initialize arrays
+        M_out  = 1*Pt_in/Pt_in
+        Ptr   = 1*Pt_in/Pt_in
+
+        # Make i_rayleigh the size of output arrays
+        i_rayleigh = Pt_in < 2*Pt_in
+        
+        # Isentropic decceleration through divergent nozzle
+        Mach[i_rayleigh]    = fm_solver(ar,Mach[i_rayleigh],gamma)  
+        
+        # Determine max stagnation temperature to thermally choke flow                                     
+        Tt4_ray = Tt_in*(1+gamma*Mach**2)**2/((2*(1+gamma)*Mach**2)*(1+(gamma-1)/2*Mach**2)) 
+
+        # Checking if Tt4 is limited by Rayleigh
+        i_low = Tt4_ray <= Tt4
+        i_high = Tt4_ray > Tt4
+        
+        # Choose Tt4 for fuel calculations
```

Comment:
As an input?

Reply:
I don't know. Please add your changes to make it happen.

Reply:
Not exactly sure how to change/what it is doing.